### PR TITLE
@W-14246983: [Android][Hybrid] Hybrid Templates Use Maven Central MSDK Dependencies By Default

### PR DESCRIPTION
**Pull request by JohnsonEricAtSalesforce**
🎸 _Ready For Review_ 🥁

  This updates the Salesforce Mobile SDK Cordova plugin such that generated Android apps will use a published version of MSDK by default from Maven Central (Or otherwise from the app's repositories configuration).  `update.sh` will still support installing MSDK locally to the app as before, plus conditional logic in the app's build scripts will use the local MSDK in a composite build as well.

### CODE REVIEW

File: tools/postinstall-android.js [hunk 1]
```diff
@@ -65,13 +65,16 @@ const pluginRoot = path.join('plugins', 'com.salesforce');
 const libProjectRoot = path.join('plugins', 'com.salesforce', 'src', 'android', 'libs');
 const appProjectRoot = path.join('platforms', 'android');
 
-console.log('Moving Salesforce libraries to the correct location');
+console.log('Moving Salesforce Mobile SDK for Android Gradle workspace to the correct location');
 shelljs.cp('-R', path.join(libProjectRoot, 'mobile_sdk'), appProjectRoot);
```
Comment (JohnsonEricAtSalesforce) [inline tools/postinstall-android.js:69]: Note that the copy command, with `-R`, will silently ignore the `mobile_sdk` directory if it is not present.  That's the default case if `update.sh` hasn't been run, which sets the app to use the Maven Central MSDK artifacts.

File: tools/postinstall-android.js [hunk 2]
```diff
@@ -65,13 +65,16 @@ const pluginRoot = path.join('plugins', 'com.salesforce');
...
 
 console.log('Fixing root level Gradle file for the generated app');
 replaceTextInFile(path.join(appProjectRoot, 'settings.gradle'), "include \":CordovaLib\"", "");
 
-shelljs.echo("includeBuild(new File(\"mobile_sdk\"))").toEnd(path.join(appProjectRoot, 'settings.gradle'));
+shelljs.echo("def salesforceMobileSdkRoot = new File('mobile_sdk');").toEnd(path.join(appProjectRoot, 'settings.gradle'));
+shelljs.echo("if (salesforceMobileSdkRoot.exists()) {").toEnd(path.join(appProjectRoot, 'settings.gradle'));
+shelljs.echo("  includeBuild(salesforceMobileSdkRoot)").toEnd(path.join(appProjectRoot, 'settings.gradle'));
```
Comment (JohnsonEricAtSalesforce) [inline tools/postinstall-android.js:76]: Now the `includeBuild` for MSDK composite builds is conditional on the existence of the root directory.

File: tools/postinstall-android.js [hunk 3]
```diff
@@ -88,7 +91,7 @@ if (data.indexOf("SalesforceHybrid") < 0)
     const oldAndroidDepTree = "android {";
     const newAndroidDepTree = "android {\n\tpackagingOptions {\n\t\texclude 'META-INF/LICENSE'\n\t\texclude 'META-INF/LICENSE.txt'\n\t\texclude 'META-INF/DEPENDENCIES'\n\t\texclude 'META-INF/NOTICE'\n\t}";
     replaceTextInFile(path.join(appProjectRoot, 'app', 'build.gradle'), oldAndroidDepTree, newAndroidDepTree);
-    const newLibDep = "api 'com.salesforce.mobilesdk:SalesforceHybrid'";
+    const newLibDep = "api 'com.salesforce.mobilesdk:SalesforceHybrid:11.0.1'";
```
Comment (JohnsonEricAtSalesforce) [inline tools/postinstall-android.js:94]: And, here we have a version for the app to use from Maven Central.  The version is ignored when the composite build is present.

Comment (wmathurin) [reply tools/postinstall-android.js:94]: We should modify https://github.com/forcedotcom/SalesforceMobileSDK-CordovaPlugin/blob/dev/setversion.sh to update that line. We run setversion.sh during releases to update version numbers that are in various files.

File: setversion.sh [hunk 4]
```diff
@@ -42,6 +42,13 @@ update_package_json ()
     gsed -i "s/\"version\":.*\"[^\"]*\"/\"version\": \"${version}\"/g" ${file}
 }
 
+update_tools_postinstall_android_sh ()
+{
+    local file=$1
+    local version=$2
+    gsed -i "s/\(com.salesforce.mobilesdk:SalesforceHybrid:\)[0-9]\+.[0-9]\+.[0-9]\+/\1$version/g" ${file}
```
Comment (JohnsonEricAtSalesforce) [inline setversion.sh:49]: @wmathurin - Here's an update to `setversion.sh` to update the MSDK version used in the template app's Gradle dependencies.

**Review by wmathurin — APPROVED** [review]

**Pull request merged**